docs/BR.md

```diff
@@ -309,7 +309,7 @@ The Definitions found in the CA/Browser Forum's Network and Certificate System S
 
 **Country**: Either a member of the United Nations OR a geographic region recognized as a Sovereign State by at least two UN member nations.
 
-**Cross-Certified Subordinate CA Certificate**: A certificate that is used to establish a trust relationship between two Root CAs.
+**Cross-Certified Subordinate CA Certificate**: A certificate that is used to establish a trust relationship between two CAs.
```

Comment:
Justification for change (for other readers):

```7.1.2.2 says that it applies when creating a cross-sign for either an existing Root CA Certificate or an existing Subordinate CA Certificate. However, the definition of "Cross-Certificated Subordinate CA Certificate" in Section 1.6.1 still just says that it establishes trust "between two Root CAs". I believe the definition should be updated to indicate that it establishes trust between any two CAs, not just between two Root CAs.```

My opinion: I agree, and this also offers better alignment with [RFC 5280](https://datatracker.ietf.org/doc/html/rfc5280#:~:text=Cross%2Dcertificates%20are%0A%20%20%20CA%20certificates%20in%20which%20the%20issuer%20and%20subject%20are%20different%0A%20%20%20entities.%20%20Cross%2Dcertificates%20describe%20a%20trust%20relationship%20between%0A%20%20%20the%20two%20CAs.)...

```Cross-certificates are
   CA certificates in which the issuer and subject are different
   entities.  Cross-certificates describe a trust relationship between
   the two CAs.  ```

Reply:
I agree with you both that the current definition of "Cross-certified" in the BRs is problematic as it does not align with the RFC 5280 definition.

However, this change should be weighed carefully, as there are explicit carve-outs/allowances for Root<-Root cross-certs in these profiles. For example, I believe this modification would allow any SubCA to not contain the EKU extension.

Reply:
Hmm, that's a good point. In 1.8.6, the language is:

> For Cross Certificates that share a Subject Distinguished Name and Subject Public Key with a Root Certificate operated in accordance with these Requirements, this extension MAY be present.

So the carve-out is based on it both meeting the criteria for a Cross Certificate **and** sharing a name+key with a Root Certificate. But in this profiles branch, the carve-out is based solely on if the Cross-Certified Subordinate CA is operated internally:

> The acceptable extensions and the requirements for those extensions in a Cross-Certified Subordinate CA vary based on whether or not the Subordinate CA is issued to and operated by the same organization as the Issuing CA or an Affiliate of the Issuing CA organization.

I'm sorry that I haven't followed the development of this profiles ballot closely enough; I'm sure this has been discussed previously. What's the reasoning for this change from "matches a Root" to "is internally operated" for making extKeyUsage a SHOULD instead of a MUST?

Reply:
This is a bit of a mess.  The key factor here is whether the cross certificate crosses an organizational boundary or not.  There's an obvious difference between one organization cross signing CAs internally, and cross signing the CA certificate of another organization, root or otherwise.  There's also the problem we never fixed of the difference between CA Organizations and CA Certificates which always makes reading these things a nightmare.  Is an internally cross sign a CC-S-CA-C?  It depends on whether you read CA=CAC or CA=CAO.  Ick.

The "cross sign between roots" language obviously was written by someone a while back who wasn't thinking about internal cross signs or cross signing ICAs, which honestly was the way most people were thinking back when most of this was written.  We need to be a bit careful to make sure the updates in this area are done carefully.

Reply:
At the end of the day, the new Cross-Certified Subordinate CA Certificate Profile (7.1.2.2) already says

> This Certificate Profile MAY be used when issuing a CA Certificate using the same Subject Name and Subject Public Key Information as one or more existing CA Certificate(s), whether a Root CA Certificate or Subordinate CA Certificate.

Therefore the bit that @CBonnell objected to (allowing an internally-controlled SubCA to not contain the EKU extension) is _already_ the case in the Profiles ballot. My change to the definition up here doesn't affect that.

If we want to address that, then let's do so, but in the mean time can we harmonize the definition of Cross-Certified Subordinate CA Certificate with the explicit language of the section which concerns it?

Reply:
@aarongable, that's a great call-out, as the allowance on excluding the EKU in a SubCA whose name and SPKI do not correspond to a Root is seemingly a regression in this branch. Additionally, it would still run afoul of MozPol.

IIRC, the intent of the "whether a Root CA Cert or SubCA Cert" was to encompass issuance where the Root CA signs an externally-operated subCA. In this case, it is deemed desirable to enforce EKU constraints on the external entity's CA. As you pointed out, this introduced a seeming regression/defect in the proposed BR language.

Reply:
By my reading of BR 1.8.6, there are actually three classes of Cross Certificates when it comes to the extKeyUsage extension:
1) matches an existing Root, internally operated: SHOULD have extKeyUsage, MAY contain anyExtendedKeyUsage
2) matches an existing Root, externally operated: SHOULD have extKeyUsage, MUST NOT contain anyExtendedKeyUsage
3) all others: MUST have extKeyUsage, MUST NOT contain anyExtendedKeyUsage

The current version of the Profiles ballot only has two classes:
1) internally operated: SHOULD have extKeyUsage, MAY contain anyExtendedKeyUsage
2) externally operated: MUST have extKeyUsage, MUST NOT contain anyExtendedKeyUsage

So I _think_ the appropriate thing to do here is to
1) Move forward with my change to the definition here; and
2) Update Section 7.1.2.2.3 to re-establish this three-way decision point, probably by reintroducing 1.8.6's "share a Subject Distinguished Name and Subject Public Key with a Root Certificate operated in accordance with these Requirements" language

Reply:
I attempted to [trace](https://docs.google.com/document/d/17EIAsQRtsPIGmySA78NXqVzF2xtRrwmKlFhZZ4aAPUY/edit?resourcekey=0-fZ4X9wve7FBeTX9wXBhZOA) Version [1.8.6](https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.8.6.pdf) of the BRs, the [original](https://github.com/cabforum/servercert/compare/2c63814fa7f9f7c477c74a6bfbeb57e0fcc5dd5b..0689ba59dbad9f5d2a5269051e5e0d0d1a25f3f6) set of proposed SC-062 changes, and those also [suggested](https://github.com/cabforum/servercert/pull/418) by @aarongable.

In summary, I agree with Aaron and think changes could be made to ensure consistency with the existing BRs for externally operated scenarios (“_matches an existing Root, externally operated: SHOULD have extKeyUsage, MUST NOT contain anyExtendedKeyUsage_”).

**Suggested updates for consistency with 1.8.6 (additive to Aaron’s [proposed changes](https://github.com/cabforum/servercert/pull/418)):**

- 7.1.2.2.3 (Cross-Certified Subordinate CA Extensions):
     - Update the third table describing “the extKeyUsage extension when the Subordinate CA is operated by an entity that is not the Issuing CA or an Affiliate of the Issuing CA” to change extKeyUsage *MUST* to *SHOULD*  

- 7.1.2.2.5 (Extended Key Usage - Restricted Cross-Certified CA):
     - Add clarification: “Restricted Cross-Certified CA Certificates that will be used to issue TLS certificates MUST contain the key purpose id-kp-serverAuth.”

This would allow Cross Certificates issued to externally operated roots to optionally include EKU (consistent with Version 1.8.6), while also constraining Cross Certificates to externally operated ICAs that issue TLS certificates to minimally require serverAuth.

@aarongable do you see it the same way?


**General comment**: the Chrome Root Program policy requires hierarchies accepted into the Chrome Root Store after 9/1/2022 (program launch date) to be [dedicated to TLS use cases](https://www.chromium.org/Home/chromium-security/root-ca-policy/#4-dedicated-tls-pki-hierarchies). The absence of EKU in a subordinate CA certificate chaining to a root added to the Chrome Root Store after 9/1/2022 would be a violation of the Chrome Root Program Policy. 

While we prefer extKeyUsage as a *MUST* in all non-root CA certificates, not all CAs following the BRs intend to be included in the Chrome Root Store.

Reply:
**Update:** As I re-read my last comment, I now realize I missed something.

**BRs Version 1.8.6, 7.1.2.2 G:** 
- "For Cross Certificates that share a Subject Distinguished Name and Subject Public Key with a Root Certificate operated in accordance with these Requirements, this extension [EKU] MAY be present."
- (continues) "This extension MAY contain the anyExtendedKeyUsage [RFC5280] usage, if the Root Certificate(s) associated with this Cross Certificate are operated by the same organization as the issuing Root Certificate."

**I interpret this to imply**: "Cross Certificates issued to Root CAs NOT operated by the same organization as the issuing Root Certificate MUST NOT contain the anyExtendedKeyUsage [RFC5280] usage [or the equivalent of not defining any EKUs]."

To that end, I think my comments above re: changes to 7.1.2.2.3 (Cross-Certified Subordinate CA Extensions) are incorrect - and the only proposed changes should be to 7.1.2.2.5, now suggested as follows:

**7.1.2.2.5 (Extended Key Usage - Restricted Cross-Certified CA):**
- Remove "If present" from "If present, the Extended Key Usage extension MUST only contain key usage purposes for which the Issuing CA has verified the Cross-Certified Subordinate CA is authorized to assert."
- Add clarification: “Restricted Cross-Certified CA Certificates that will be used to issue TLS certificates MUST contain the key purpose id-kp-serverAuth.”

Reply:
@ryancdickson , regarding:

> The absence of EKU in a subordinate CA certificate chaining to a root added to the Chrome Root Store after 9/1/2022 would be a violation of the Chrome Root Program Policy

What is the purpose of requiring an EKU in CA Certificates if the entire hierarchy is restricted to TLS (at the Trust Anchor level)? It adds bytes on the wire for each TLS handshake and brings no security benefit. If I remember correctly, the purpose of adding that EKU was to restrict applicable policies in multi-purpose hierarchies. If the hierarchy is TLS-only, I don't see any reason to require this EKU for CA Certificates.

When creating cross-certificates to multi-purpose Roots, then it makes more sense to further restrict via EKU.

Reply:
@dzacharo

The statement “If the hierarchy is restricted to TLS (at the Trust Anchor level)” might assume there is a capability to distinguish the hierarchical restriction by the verifier outside of the content included in the root CA certificate. For Application Software Suppliers, or their corresponding customers, that do not or can not use metadata, I’m not aware of any content within the root CA certificate alone that would strongly signal how the hierarchy should be constrained (to be clear, I do not consider including “TLS” in the subject DN a strong signal).  

Regardless of specific Application Software Supplier implementations, it would seem the BRs (existing today and the proposed update) agree with our program policy that EKU is required on subordinate CA certificates with one exception (cross certificates whose represented issuer and subject CAs are both roots and operated by the same organization - described below).

BRs Section 1.8.6, 7.1.2.2 (Subordinate CA Certificate), Part G: "_For all other Subordinate CA Certificates, including Technically Constrained Subordinate CA Certificates: This extension MUST be present and SHOULD NOT be marked critical._"

The only subordinate CA certificates described prior to the above statement are:
- cross certificates whose represented issuer and subject CAs are (A) both roots and (B) operated by the same organization; and
- cross certificates whose represented issuer and subject CAs are (A) both roots and (B) operated by different organizations. 

As to why I think cross certificates whose represented issuer and subject CAs are (A) both roots and (B) operated by different organizations require an explicit EKU:

(1) BRs Section 1.8.6, 7.1.2.2 (Subordinate CA Certificate), Part G states: "This extension MAY contain the anyExtendedKeyUsage [RFC5280] usage, if the Root Certificate(s) associated with this Cross Certificate are operated by the same organization as the issuing Root Certificate."

(2) If we imagined three identical certificates with only the following differences: 

- Certificate Number 1: expresses no EKU
- Certificate Number 2: expresses an EKU with a purpose of anyExtendedKeyUsage 
- Certificate Number 3: expresses the set of all known EKUs (if this was possible)

How would an RFC 5280-compliant certificate verifier (if we can use such a term) treat them? If EKU (Section 4.2.1.12) of 5280 “indicates one or more purposes for which the certified public key may be used, in addition to or in place of the basic purposes indicated in the key usage extension” my interpretation is that they should be treated the same (i.e., they are “operationally” equivalent). 

(3) only cross certificates where issuer/subject are operated by the same organization are permitted to include anyEKU (i.e., number 1), or by transitive property, express no EKU (i.e., number 2).

(4) If it was intended that cross certificates issued to external CAs did not require an explicit set of EKUs, the same exception allowing anyEKU described above (i.e., number 1) would have been made.

[1] Footnote on Pg. 75 of [Version 1.8.6 of the BRs](https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.8.6.pdf): "While RFC 5280, Section 4.2.1.12 notes that this extension will generally only appear within end-entity certificates, these Requirements make use of this extension to further protect relying parties by limiting the scope of CA Certificates, as implemented by a number of Application Software Suppliers."

Hopefully this also more clearly describes the position in my [earlier](https://github.com/cabforum/servercert/pull/418#discussion_r1100357819) comment.

Reply:
I am not challenging what the current BRs say or what the current Root Store policies says. My comment is about native TLS hierarchies. Non-Browser implementations don't check the EKU at the CA level anyway. In the future, if there were native TLS hierarchies, including the EKU extension in a CA doesn't "further protect relying parties" because they are protected by the actual design of the "nativeness" of the hierarchy.

So, if a CA knows that it manages such a native TLS hierarchy, perhaps it would make sense to consider removing that requirement in the future at the CA level, because it doesn't add any benefits, but leave it for hierarchies that are mixed.

Anyway, my comment was mostly about Google Chrome's Policy which is TLS-only and already pushing for native TLS hierarchies.